clippy_lints/src/lib.rs

```diff
@@ -494,6 +494,9 @@ pub struct LintInfo {
     /// e.g. `clippy_lints/src/absolute_paths.rs#43`
     pub location: &'static str,
     pub version: Option<&'static str>,
+    /// Any optional subgroups this lint belongs to, e.g. `perf_pedantic`
+    /// (see `book/src/lint_subgroups.md`).
+    pub subgroups: &'static [&'static str],
```

Comment:
Now that I think about it, I don't think we even really need to store this. It's only used by `cargo dev update_lints` and that gets all the lint info by parsing the source code anyway, so we could just ignore it in the macro. Posting this comment here as a reminder so I don't forget.